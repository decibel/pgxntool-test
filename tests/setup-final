#!/bin/bash

trap 'echo "ERROR: $BASH_SOURCE: line $LINENO" >&2' ERR
set -o errexit -o errtrace -o pipefail
#set -o xtrace -o verbose

if [ "$1" == "-v" ]; then
  verboseout=1
  shift
fi
BASEDIR=`cd ${0%/*}; pwd`

. $BASEDIR/../.env
. $TOPDIR/lib.sh
cd $TEST_REPO

EXTENSION_NAME=pgxntool-test # TODO: rename to something less likely to conflict

# We do this here instead of in the setup test to make sure none of our new stuff gits changed
out "Run setup.sh again to verify it doesn't over-write things"
pgxntool/setup.sh
git diff --exit-code

out Copy stuff from template to where it normally lives
cp -R t/* .

out Add extension to deps.sql
quote='"'
sed -i '' -e "s/CREATE EXTENSION \.\.\..*/CREATE EXTENSION ${quote}$EXTENSION_NAME${quote};/" test/deps.sql

check_log

# vi: expandtab sw=2 ts=2
