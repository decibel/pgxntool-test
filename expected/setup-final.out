# Run setup.sh again to verify it doesn't over-write things
.gitignore already exists
META.in.json already exists
Makefile already exists
make[1]: `META.json' is up to date.
deps.sql already exists
On branch master
Your branch is ahead of 'origin/master' by 4 commits.
  (use "git push" to publish your local commits)
nothing to commit, working tree clean
If you won't be creating C code then you can:

rmdir src

If everything looks good then

git commit -am 'Add pgxntool (https://github.com/decibel/pgxntool/tree/release)'
# Copy stuff from template to where it normally lives
# Add extension to deps.sql
